src/API/OnboardingTasks.php

```diff
@@ -715,170 +767,252 @@ function( $has_installed, $bundle ) {
-							: null,
-						'actionLabel' => __( "Let's go", 'woocommerce-admin' ),
-						'isComplete'  => count( \WC_Shipping_Zones::get_zones() ) > 0,
-						'isVisible'   => in_array( 'physical', $product_types, true ) ||
-							count(
-								wc_get_products(
-									array(
-										'virtual' => false,
-										'limit'   => 1,
-									)
-								)
-							) > 0,
-						'time'        => __( '1 minute', 'woocommerce-admin' ),
-					),
-					array(
-						'id'         => 'marketing',
-						'title'      => __( 'Set up marketing tools', 'woocommerce-admin' ),
-						'content'    => __(
-							'Add recommended marketing tools to reach new customers and grow your business',
-							'woocommerce-admin'
-						),
-						'isComplete' => $has_installed_marketing_extensions,
-						'isVisible'  => Features::is_enabled( 'remote-free-extensions' ) && count( $marketing_extension_bundles ) > 0,
-						'time'       => __( '1 minute', 'woocommerce-admin' ),
-					),
-					array(
-						'id'          => 'appearance',
-						'title'       => __( 'Personalize my store', 'woocommerce-admin' ),
-						'content'     => __(
-							'Add your logo, create a homepage, and start designing your store.',
-							'woocommerce-admin'
-						),
-						'actionLabel' => __( "Let's go", 'woocommerce-admin' ),
-						'isComplete'  => get_option( 'woocommerce_task_list_appearance_complete' ),
-						'isVisible'   => true,
-						'time'        => __( '2 minutes', 'woocommerce-admin' ),
-					),
+						)
+					) > 0,
+				'time'        => __( '1 minute', 'woocommerce-admin' ),
+			),
+			array(
+				'id'         => 'marketing',
+				'title'      => __( 'Set up marketing tools', 'woocommerce-admin' ),
+				'content'    => __(
+					'Add recommended marketing tools to reach new customers and grow your business',
+					'woocommerce-admin'
 				),
+				'isComplete' => $has_installed_marketing_extensions,
+				'isVisible'  => Features::is_enabled( 'remote-free-extensions' ) && count( $marketing_extension_bundles ) > 0,
+				'time'       => __( '1 minute', 'woocommerce-admin' ),
+			),
+			array(
+				'id'          => 'appearance',
+				'title'       => __( 'Personalize my store', 'woocommerce-admin' ),
+				'content'     => __(
+					'Add your logo, create a homepage, and start designing your store.',
+					'woocommerce-admin'
+				),
+				'actionLabel' => __( "Let's go", 'woocommerce-admin' ),
+				'isComplete'  => get_option( 'woocommerce_task_list_appearance_complete' ),
+				'isVisible'   => true,
+				'time'        => __( '2 minutes', 'woocommerce-admin' ),
 			),
 		);
 
-		return rest_ensure_response( apply_filters( 'woocommerce_admin_onboarding_tasks', $tasks ) );
+		$snoozed_tasks     = get_option( 'woocommerce_task_list_remind_me_later_tasks' );
+		$tasks_with_snooze = array_map(
+			function ( $task ) use ( $snoozed_tasks ) {
+				$is_snoozed        = in_array( $task['id'], array_keys( $snoozed_tasks ), true );
+				$task['isSnoozed'] = $is_snoozed;
```

Comment:
Should this check to see if the snooze time has already passed?

Reply:
Perhaps, I'd assumed this option was already being managed via a CRON operation, but I'm glad to address this. I'm not sure what action we'd want to take in this case?

Reply:
I don't think this needs to be managed by cron.  If we store a timestamp with the date a task is snoozed until, we can compare that we when retrieve timestamps and check wether or not they're still snoozed:

```php
$task[ 'isSnoozed' ] = $snoozed_until > now();
```